avm/res/sql/server/main.bicep

```
@@ -799,3 +798,96 @@ type elasticPoolPropertyType = {
   @description('Optional. Whether or not this elastic pool is zone redundant, which means the replicas of this elastic pool will be spread across multiple availability zones.')
   zoneRedundant: bool?
 }
+
+@export()
+type encryptionProtectorType = {
+  @description('Required. The name of the server key.')
+  serverKeyName: string
+
+  @description('Optional. The encryption protector type like \'ServiceManaged\', \'AzureKeyVault\'.')
+  serverKeyType: 'ServiceManaged' | 'AzureKeyVault'?
+
+  @description('Optional. Key auto rotation opt-in flag. Either true or false.')
```

```suggestion
  @description('Optional. Key auto rotation opt-in flag.')
```
A given for type boolean.